lib/shared/bucketing-assembly-script/assembly/types/target.ts

```diff
@@ -231,18 +234,46 @@ export class CustomDataFilter extends UserFilter {
         this.dataKey = getStringFromJSON(filter, 'dataKey')
         this.dataKeyType = isValidString(filter, 'dataKeyType', validDataKeyTypes)
     }
+    stringify(): string {
+        const json = new JSON.Obj()
+        if (this.dataKey) {
+            json.set('dataKey', this.dataKey)
+        }
+        if (this.dataKeyType) {
+            json.set('dataKeyType', this.dataKeyType)
+        }
+        if (this.type) {
+            json.set('type', this.type)
+        }
+        if (this.subType) {
+            json.set('subType', this.subType)
+        }
+        if (this.comparator) {
+            json.set('comparator', this.comparator)
+        }
+        if (this.values) {
+            json.set('values', this.values)
+        }
+        return json.stringify()
+    }
 }
 
-function initializeFilterClass(filter: JSON.Obj): AudienceFilterOrOperator {
+function initializeFilterClass(filter: JSON.Obj): AudienceFilter | null {
+    if (isValidStringOptional(filter, 'operator', validAudienceOperators)){
+        return null
+    }
+
     if (getStringFromJSONOptional(filter, 'type') === 'user') {
         if (getStringFromJSONOptional(filter, 'subType') === 'customData') {
             return new CustomDataFilter(filter)
         }
         return new UserFilter(filter)
     } else if (getStringFromJSONOptional(filter, 'type') === 'audienceMatch'){
         return new AudienceMatchFilter(filter)
+    } else if (getStringFromJSONOptional(filter, 'type') === 'all'){
+        return new AudienceFilter(filter)
     } else {
-        return new AudienceFilterOrOperator(filter)
+        throw new Error('Invalid target filter structure. Empty filters detected.')
```

Comment:
wont this now throw whenever a new type is added that isnt covered by the if statement above?

Reply:
doesnt that contradict your earlier comment about how the AudienceOperatorOrFilter class should throw if filterClass and operatorClass are both null? If I dont do it this way then it will always produce a filter and never throw

Reply:
well what it used to do is generate a valid "AudienceFilter" class as long as there was a type of any kind set. I was more thinking that it should basically be impossible for both those fields to be null, either you'd have an operator or an audience filter set. But the audience filter might be "invalid".

We have downstream code which handles for audience filters that have an unknown type, which is why it needs to be initialized that way

Reply:
yeah, so basically its impossible to throw in this class if theyre both null, because if the filter is ever null that means we arent forwards compatible in adding new types

Reply:
yeah but the filter shouldnt be null if the type is unknown, it should be `AudienceFilter`